src/main/java/io/r2dbc/mssql/message/token/EnvChangeToken.java

```diff
@@ -69,13 +69,26 @@ public static EnvChangeToken decode(ByteBuf buffer) {
 
         EnvChangeType envChangeType = EnvChangeType.valueOf(type);
 
-        int newValueLen = envChangeType.toByteLength(Decode.asByte(buffer));
-        byte[] newValue = new byte[newValueLen];
-        buffer.readBytes(newValue);
+        byte[] newValue;
+        byte[] oldValue;
 
-        int oldValueLen = envChangeType.toByteLength(Decode.asByte(buffer));
-        byte[] oldValue = new byte[oldValueLen];
-        buffer.readBytes(oldValue);
+        if (envChangeType == EnvChangeType.Routing) {
```

Comment:
We should probably leave a comment for both branches saying that `Routing` has a structured content and other tokens just report old/new values.

Reply:
Yes. I have left a comment there now.